File: C#/Droppable/Droppable/Droppable/IO/Engines/EngineMP3raid.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Net;

using Droppable.IO.Utils;

namespace Droppable.IO.Engines
{
    public class EngineMP3raid : Engine
    {
        public EngineMP3raid() :
            base("MP3raid", "http://www.mp3raid2.com/", "download/", 1)
        { }

        public override string Search(SongInfo info)
        {
            var realSongInfo = info;
            realSongInfo.ArtistName = realSongInfo.ArtistName.Replace("+", " ");
            realSongInfo.SongName = realSongInfo.SongName.Replace("+", " ");

            var pageContent = new DroppableWebClient().DownloadStringDispose(base.GetBaseURL() + base.GetSearchURL() + realSongInfo.ToString() + ".html");

            var foundSongs = pageContent.Explode("<div class='index1'>");

            var loopIndex = base.GetBaseIndex();

            while (true)
            {
                if (loopIndex >= foundSongs.Length)
                    break;

                var currentSongId = foundSongs[loopIndex].Explode("<a href='javascript:;' class='dl' id='")[1].Explode("'>")[0];
                currentSongId = currentSongId.Replace("dl", "");

                var currentSongPageContent = new DroppableWebClient().DownloadStringDispose(base.GetBaseURL() + "search/ddl/" + currentSongId + "/" + realSongInfo.ArtistName + ".html");

                var downloadURL = currentSongPageContent.Explode("<b>Location:</b></td><td style='word-wrap:break-word;'>")[1].Explode("</td></tr>")[0];

                if (SharedProperties.Settings.FastSearch || MP3Utils.CheckMP3(info, downloadURL))
                {
                    return downloadURL;
                }

                loopIndex++;
            }

            return null;
        }

        public override string SearchDirty(string searchQuery)
        {
            return null; // this engine doesn't support dirty search
        }
    }
}

File: Droppable/bin/_new_/EngineMP3raid.php
<?php

	class EngineMP3raid extends BaseEngine
	{
		protected $name = "MP3raid";
		protected $baseURL = "http://www.mp3raid.ca/";
		protected $searchURL = "download/";
		protected $baseIndex = 1;

		public function Search($songInfo, $an, $sn)
		{
			$realArtistName = str_replace("+", "_", $an);
			$realSongName = str_replace("+", "_", $sn);
			$realSongInfo = str_replace("+", "_", $songInfo);

			$pageContent = file_get_contents($this->GetBaseURL() . $this->searchURL . $realSongInfo . ".html");

			$foundSongs = explode('<div class=\'index1\'>', $pageContent);

			$loopIndex = $this->baseIndex;

			while (true)
			{
				if ($loopIndex >= count($foundSongs))
					break;

				$currentSongId = explode('\'>', explode('<a href=\'javascript:;\' class=\'dl\' id=\'', $foundSongs[$loopIndex])[1])[0];
				$currentSongId = str_replace("dl", "", $currentSongId);

				$currentSongPageContent = file_get_contents('http://www.mp3raid.ca/search/ddl/' . $currentSongId . '/' . $realArtistName . '.html');

				$downloadURL = explode('</td></tr>', explode('<b>Location:</b></td><td style=\'word-wrap:break-word;\'>', $currentSongPageContent)[1])[0];

				if (true)
				{
					return $downloadURL;
				}

				$loopIndex++;
			}

			return null;
		}

		public function ResolveDownloadURL($url)
		{
			return $url;
		}
	}

?>